### DISCUSSION TITLE
[QUESTION] How to deploy longhorn in a subset of nodes and offer service for all nodes in k8s cluster

### DISCUSSION THREAD

**Discussion by litao3rd** [longhorn/longhorn#6896] [General]
## Question

Hello, everyone.

I have a small Kubernetes cluster with 8 nodes. Out of these, 5 nodes are running the CentOS 7 distribution, and the remaining 3 nodes are using the Ubuntu 22.04 distribution. My goal is to install Longhorn exclusively on the Ubuntu nodes because they meet all the requirements for deploying Longhorn. However, the other 5 nodes do not meet these requirements, but they still need to access Longhorn services when using PVC.

Initially, I thought that by installing the core components on the Ubuntu nodes and other interface components such as the CSI plugin on all user nodes, I could meet my requirements. Unfortunately, this approach did not work as expected.

I have consulted [this documentation](https://longhorn.io/kb/tip-only-use-storage-on-a-set-of-nodes). According to the documentation, if I install Longhorn components only on specific nodes, it will only support workloads running on those nodes to access Longhorn services.

However, I noticed that the documentation is based on an older version. I am curious to know if there have been any improvements that might help resolve this issue more effectively.

## Environment

- Longhorn version: 1.5.0
- Kubernetes version: 1.26.8
- Node config
   - OS type and version: ubuntu 22.04 /  CentOS 7
   - Kernel version: 5.15 / 3.10
   - CPU per node: 4 / 32
   - Memory per node: 16G / 64G
   - Disk type: sata 
   - Network bandwidth and latency between the nodes: not test
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):  kvm

## Additional context

None

**Comment by DamiaSan** [comment]
If you follow the steps described at [Tell Longhorn to create a default disk on a specific set of nodes](https://longhorn.io/kb/tip-only-use-storage-on-a-set-of-nodes/#tell-longhorn-to-create-a-default-disk-on-a-specific-set-of-nodes), the result will be
_workloads that use Longhorn volumes can run on any nodes. Longhorn only uses storage on node-1, node-2, and node-3 for replica scheduling_. That is exactly what you need, right?

**Reply by litao3rd** [reply]
Yep. Thanks for your reply. That's make it. And following the documentation, I  discovered that the components, including longhorn-manager, longhorn-csi-plugin, instance-manager and engine-image-ei, will be deployed as daemonset on all nodes. I was curious does it work as expected.
